chromadb/auth/fastapi.py

```diff
@@ -168,41 +174,50 @@ def wrapped(*args: Any, **kwargs: Dict[Any, Any]) -> Any:
-                        if isinstance(resource, AuthzResource)
-                        else resource.to_authz_resource(**_dynamic_kwargs)
-                    )
-                    _context = AuthorizationContext(
-                        user=AuthzUser(
-                            id=request.state.user_identity.get_user_id()
-                            if hasattr(request.state, "user_identity")
-                            else "Anonymous",
-                            tenant=request.state.user_identity.get_user_tenant()
-                            if hasattr(request.state, "user_identity")
-                            else DEFAULT_TENANT,
-                            attributes=request.state.user_identity.get_user_attributes()
-                            if hasattr(request.state, "user_identity")
-                            else {},
-                        ),
-                        resource=_resource,
-                        action=_action,
-                    )
-
-                    if _provider:
-                        a_authz_responses.append(_provider.authorize(_context))
-                if not any(a_authz_responses):
-                    raise AuthorizationError("Unauthorized")
+            if not request:
+                return f(*args, **kwargs)
+
+            _provider = authz_provider.get()
+            a_list: List[Union[str, AuthzAction]] = []
+            if not isinstance(action, List):
+                a_list = [action]
+            else:
+                a_list = cast(List[Union[str, AuthzAction]], action)
+            a_authz_responses = []
+            for a in a_list:
+                _action = a if isinstance(a, AuthzAction) else AuthzAction(id=a)
+                _resource = (
+                    resource
+                    if isinstance(resource, AuthzResource)
+                    else resource.to_authz_resource(**_dynamic_kwargs)
+                )
+                _context = AuthorizationContext(
+                    user=AuthzUser(
+                        id=request.state.user_identity.get_user_id()
+                        if hasattr(request.state, "user_identity")
+                        else "Anonymous",
+                        tenant=request.state.user_identity.get_user_tenant()
+                        if hasattr(request.state, "user_identity")
+                        else DEFAULT_TENANT,
+                        attributes=request.state.user_identity.get_user_attributes()
+                        if hasattr(request.state, "user_identity")
+                        else {},
+                    ),
+                    resource=_resource,
+                    action=_action,
+                )
+
+                if _provider:
+                    a_authz_responses.append(_provider.authorize(_context))
+            if not any(a_authz_responses):
+                raise AuthorizationError("Unauthorized")
+            if overwrite_singleton_tenant_database_access_from_auth:
```

Comment:
nit: Add a comment explaining this since this whole function is a bit obtuse.

Reply:
Done